### PR TITLE
Workflow to build and test on ARM64 for 4.x branch

### PR DESCRIPTION
**Pull request by fengyuentau**
Black list of tests can be found [here](https://github.com/opencv-cn-ci-bot/opencv-cn-pipelines/blob/master/precommit/ubuntu-20.04-arm64.groovy)

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: .github/workflows/PR-4.x-ARM64.yaml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        ref: ${{ env.TARGET_BRANCH_NAME }}
+        fetch-depth: 0
+    - name: Merge opencv with a test branch
+      run: |
+        cd ${{ env.OPENCV_DOCKER_WORKDIR }}
+        git config --global --add safe.directory ${{ env.OPENCV_DOCKER_WORKDIR }}
+        git config user.email "opencv.ci"
+        git config user.name "opencv.ci"
+        git pull -v "https://github.com/${{ env.PR_AUTHOR_FORK }}" "${{ env.SOURCE_BRANCH_NAME }}"
+    - name: Clone opencv_contrib
```
[the remark this answers is not in the document]
Comment (fengyuentau) [reply .github/workflows/PR-4.x-ARM64.yaml:161]: Yes, the check is added.

File: .github/workflows/PR-4.x-ARM64.yaml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    - name: Accuracy:core
+      run: cd /opencv-build && xvfb-run -a bin/opencv_test_core
+    - name: Accuracy:dnn
+      run: cd /opencv-build && xvfb-run -a bin/opencv_test_dnn
+    - name: Accuracy:features2d
+      run: cd /opencv-build && xvfb-run -a bin/opencv_test_features2d
+    - name: Accuracy:flann
+      run: cd /opencv-build && xvfb-run -a bin/opencv_test_flann
+    - name: Accuracy:highgui
+      run: cd /opencv-build && xvfb-run -a bin/opencv_test_highgui
```
[the remark this answers is not in the document]
Comment (fengyuentau) [reply .github/workflows/PR-4.x-ARM64.yaml:81]: GAPI tests are disabled in the [old pipeline](https://github.com/opencv-cn-ci-bot/opencv-cn-pipelines/blob/306c133eeef2bfefc507dc9d1513e4ed4b9098ee/precommit/ubuntu-20.04-arm64.groovy#L16). Is it okay to turn it on now?

[the remark this answers is not in the document]
Comment (fengyuentau) [reply .github/workflows/PR-4.x-ARM64.yaml:83]: GAPI tests are enabled. Lets see what happens.

**Comment by fengyuentau** [comment]
@asmorkalov @asenyaev  Some G-API tests failed:
```shell
[----------] Global test environment tear-down
[==========] 20877 tests from 509 test cases ran. (1856588 ms total)
[  PASSED  ] 20854 tests.
[  FAILED  ] 23 tests, listed below:
[  FAILED  ] PatternMatchingIntegrationE2E.ChainTransformationsApplied
[  FAILED  ] PatternMatchingIntegrationE2E.ReversedChainTransformationsApplied
[  FAILED  ] MathOpTestFluid/MathOpTest.MatricesAccuracyTest/26, where GetParam() = (8UC3, 1280x720, SAME_TYPE, 0xaaaadd2a5540, DIV, true, 1, false)
[  FAILED  ] MathOpTestFluid/MathOpTest.MatricesAccuracyTest/34, where GetParam() = (8UC3, 1280x720, 8UC1, 0xaaaadd2a5540, DIV, true, 1, false)
[  FAILED  ] MathOperatorArithmeticTestFluid/MathOperatorMatScalarTest.OperatorAccuracyTest/3, where GetParam() = (8UC1, 1280x720, SAME_TYPE, 0xaaaadd31f600, AbsExact(), DIV)
[  FAILED  ] MathOperatorArithmeticTestFluid/MathOperatorMatScalarTest.OperatorAccuracyTest/7, where GetParam() = (8UC1, 1280x720, SAME_TYPE, 0xaaaadd31f600, AbsExact(), DIVR)
[  FAILED  ] MathOperatorArithmeticTestFluid/MathOperatorMatScalarTest.OperatorAccuracyTest/11, where GetParam() = (8UC1, 640x480, SAME_TYPE, 0xaaaadd31f600, AbsExact(), DIV)
[  FAILED  ] MathOperatorArithmeticTestFluid/MathOperatorMatScalarTest.OperatorAccuracyTest/15, where GetParam() = (8UC1, 640x480, SAME_TYPE, 0xaaaadd31f600, AbsExact(), DIVR)
[  FAILED  ] MathOperatorArithmeticTestFluid/MathOperatorMatScalarTest.OperatorAccuracyTest/19, where GetParam() = (8UC1, 128x128, SAME_TYPE, 0xaaaadd31f600, AbsExact(), DIV)
[  FAILED  ] MathOperatorArithmeticTestFluid/MathOperatorMatScalarTest.OperatorAccuracyTest/23, where GetParam() = (8UC1, 128x128, SAME_TYPE, 0xaaaadd31f600, AbsExact(), DIVR)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/2, where GetParam() = (8UC1, 1280x720, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/11, where GetParam() = (8UC1, 640x480, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/20, where GetParam() = (8UC1, 128x128, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/29, where GetParam() = (16SC1, 1280x720, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/38, where GetParam() = (16SC1, 640x480, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/47, where GetParam() = (16SC1, 128x128, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/56, where GetParam() = (32FC1, 1280x720, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/65, where GetParam() = (32FC1, 640x480, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] MathOperatorTestFluid/MathOperatorMatMatTest.OperatorAccuracyTest/74, where GetParam() = (32FC1, 128x128, SAME_TYPE, 0xaaaadd31f650, AbsExact(), DIV)
[  FAILED  ] Resize220_400TestCPU/ResizeTestFluid.SanityTest/0, where GetParam() = (0, 1, 220x220, (400x400, [0 x 0 from (0, 0)]), 1, 0)
[  FAILED  ] Resize220_400TestCPU/ResizeTestFluid.SanityTest/1, where GetParam() = (0, 1, 220x220, (400x400, [0 x 0 from (0, 0)]), 2, 0)
[  FAILED  ] Resize220_400TestCPU/ResizeTestFluid.SanityTest/2, where GetParam() = (0, 1, 220x220, (400x400, [0 x 0 from (0, 0)]), 3, 0)
[  FAILED  ] Resize220_400TestCPU/ResizeTestFluid.SanityTest/3, where GetParam() = (0, 1, 220x220, (400x400, [0 x 0 from (0, 0)]), 4, 0)

23 FAILED TESTS
  YOU HAVE 1677 DISABLED TESTS
```

Should we turn it on but filter out those failed?

**Comment by asenyaev** [comment]
> Should we turn it on but filter out those failed?

I would suggest to turn it off at this moment and to deep dive in this issue a bit later.

**Comment by fengyuentau** [comment]
G-API tests are disabled.

**Review by asenyaev — APPROVED** [review]
Thanks!

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**